optuna/visualization/_pareto_front.py

```diff
@@ -96,19 +97,36 @@ def objective(trial):
             If given, trials are classified into three categories: feasible and best, feasible but
             non-best, and infeasible. Categories are shown in different colors. Here, whether a
             trial is best (on Pareto front) or not is determined ignoring all infeasible trials.
+        targets:
+            A function that returns targets values to display.
+            The argument to this function is :class:`~optuna.trial.FrozenTrial`.
+
+            .. note::
+                Added in v3.0.0 as an experimental feature. The interface may change in newer
+                versions without prior notice.
+                See https://github.com/optuna/optuna/releases/tag/v3.0.0.
 
     Returns:
         A :class:`plotly.graph_objs.Figure` object.
 
     Raises:
         :exc:`ValueError`:
             If the number of objectives of ``study`` isn't 2 or 3.
+        :exc:`ValueError`:
+            If ``targets`` returns something other than sequence.
+        :exc:`ValueError`:
+            If the number of target values to display isn't 2 or 3.
```

Comment:
I got `AssertionError` instead of `ValueError` if I return four objective values. Could you check it, please?

```python
import optuna
study = optuna.create_study(directions=["minimize"] * 4)
import random
study.optimize(lambda t: [random.random() for _ in range(4)], n_trials=10)
optuna.visualization.plot_pareto_front(study, targets=lambda t: [t.values[i] for i in range(4)] )
```

Error
```
~/pfn/code/optuna-review/optuna/visualization/_pareto_front.py in _make_scatter_object(n_targets, axis_order, include_dominated_trials, trials_with_values, hovertemplate, infeasible, dominated_trials)
    367     trials_with_values = trials_with_values or []
    368 
--> 369     assert n_targets in (2, 3)
    370     marker = _make_marker(
    371         [trial for trial, _ in trials_with_values],
```

Reply:
I talked with @himkt and they mentioned that docstrings related Exceptions will be re-organized in https://github.com/optuna/optuna/pull/3429. So, I felt that this issue is out-of-scope of this PR, and we'll work on it in a follow-up PR.

Reply:
I created a tracking issue https://github.com/optuna/optuna/issues/3502.